penny_university_frontend/src/middleware/api.ts

```diff
@@ -62,20 +63,42 @@ const api: Middleware<Dispatch> = (store: MiddlewareAPI) => (next: (action: AnyA
     }
 
     const [requestType, successType, failureType] = types
-    next({ type: requestType, payload: { meta } })
+    next({type: requestType, payload: {meta}})
     const token = selectors.user.getToken(store.getState())
     return callApi(endpoint, method, payload, token).then(
       (response: any) => {
-        const metaWithPagination = response.results ? { ...meta, count: response.count, next: response.next, previous: response.previous } : meta
+        const metaWithPagination = response.results ? {
+          ...meta,
+          count: response.count,
+          next: response.next,
+          previous: response.previous
+        } : meta
         return next({
-          payload: { result: response.results || response, responseSchema, meta: metaWithPagination },
+          payload: {result: response.results || response, responseSchema, meta: metaWithPagination},
           type: successType,
         })
       },
-      (error: { message: string, status: number }) => next({
-        type: failureType,
-        payload: { message: error.message || 'An error occurred.', status: error.status, meta },
-      }),
+      async (error: any) => {
+        let errorBody = null;
+        try {
+          await error.json().then((json: any) => {
+            errorBody = camelizeKeys(json)
+          }).catch()
```

Comment:
what does `catch` do here? if it does nothing then why have it?

Reply:
I was just following what was done with the non-error response to be honest.